alexandria/dav.py

```diff
@@ -0,0 +1,163 @@
+import io
+import time
+from pathlib import Path
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.files.base import File as DjangoFile
+from django_clamd.validators import validate_file_infection
+from manabi import ManabiAuthenticator, ManabiDAVApp
+from manabi.filesystem import ManabiFileResourceMixin, ManabiProvider
+from manabi.lock import ManabiDbLockStorage
+from manabi.log import HeaderLogger
+from wsgidav.dav_error import HTTP_FORBIDDEN, DAVError
+from wsgidav.dav_provider import DAVNonCollection
+from wsgidav.dir_browser import WsgiDavDirBrowser
+from wsgidav.error_printer import ErrorPrinter
+from wsgidav.mw.debug_filter import WsgiDavDebugFilter
+from wsgidav.request_resolver import RequestResolver
+
+from alexandria.core.models import File
+
+
+class MyBytesIO(io.BytesIO):
+    """
+    Custom BytesIO.
+
+    `wsgidav.RequestServer.do_PUT` would close the file after writing. This class just
+    overrides the `close()`-method in order to prevent this, when it's called without
+    `ignore=False`.
```

Comment:
Needs more explanation - why is this needed? What would happen without it?

Reply:
Done.